Documentation/operations/upgrade.rst

```diff
@@ -333,6 +333,15 @@ Annotations:
   install --set ipv4.enabled=true``. The following section calls out specific
   values where the prefix cannot be simply dropped followed by a full
   table of old and new Helm values.
+* In case the BPF kube-proxy replacement is enabled under direct routing operation
+  as well as BPF-based masquerading, then the agent will probe the underlying
+  kernel for availability of BPF-based routing and neighbor resolution in the
```

Comment:
> the agent will probe the underlying kernel for availability of BPF-based routing and neighbor resolution

Where does this feature availability check take place?

> it also means that netfilter/iptables rules in the host namespace are not considered

Default behavioural changes like these should be featured a bit more prominently, as this one in particular significantly alters the way in which a packet moves through the kernel. This might interfere with other networking-related tools the user might have deployed, don't underestimate the amount of observability tooling built around iptables that will suddenly be disarmed for good. :slightly_smiling_face:  Right now this info is a bit buried in context.

We should lead with the most important info:
> **On kernel 5.10 and above, running the BPF kube-proxy replacement could bypass iptables in the host network namespace in some cases.**
.. and then jump into the details/context.

Alternatively, if we have many such gotchas in a release, we could consider adding a 'breaking changes' heading with distilled information to give a better overview.

Reply:
+1!

Reply:
Makes sense, I'll rewrite this bit. Keep in mind that 5.10 is not out the gate, so users are not running it at this point hence also no breakage from live-upgrade. Anyway, I'll make it a bit more clear.